src/peft/tuners/lora.py

```diff
@@ -411,7 +427,7 @@ def _prepare_lora_config(peft_config, model_config):
             peft_config.target_modules = TRANSFORMERS_MODELS_TO_LORA_TARGET_MODULES_MAPPING[model_config["model_type"]]
         return peft_config
 
-    def merge_and_unload(self):
+    def merge_and_unload(self, merge=True):
```

Comment:
Does it make sense to have a method called `merge_and_unload` with the option to have `merge=False`?

How about moving the content of this method inside a new method `_unload_and_optionally_merge` or some similar name. Then we have:

```python
def merge_and_unload(self):
    return self._unload_and_optionally_merge(merge=True)

def unload(self):
    return self._unload_and_optionally_merge(merge=False)
```

Reply:
makes sense

Reply:
Done